### PR TITLE
Pre signed url configuration

### LINKED ISSUE #91: Make pre-signed middleware configurable
Add options to restrict which http methods are allowed for pre-signed requests.
Currently we only want the clients to GET resources.

### PR DESCRIPTION
**Pull request by C0rby**
Adds an option to set allowed HTTP methods for pre-signed requests.
Closes https://github.com/owncloud/ocis-proxy/issues/91

### CODE REVIEW

**Comment by PVince81** [comment]
I thought the concern was more about checking that "OC-Verb" matches the request.

From my understanding whoever is signing the URL can define with what verb the URL is allowed to be used.

Now this PR is still nice as it would prevent signing requests to use any other verbs.

Thoughts ? @butonic

**Comment by C0rby** [comment]
Hm then I don't quite get your thought.
What exactly should be checked because 
> checking that "OC-Verb" matches the request

is done here: https://github.com/owncloud/ocis-proxy/pull/92/files#diff-87fbfdeb118753d80687b8e34bed61bfR81

But as you wrote whoever signs the URL can choose which verb/method to use.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ pkg/middleware/presigned_url.go  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-proxy/pullRequest?prid=6040115&bid=19791446)

**Comment by PVince81** [comment]
I was mostly echoing @butonic's thoughts on this and what was missing. I should have checked the code for that, sorry.

In any case, this PR seems useful anyway to make sure nothing else than GET is doable for now.

File: pkg/middleware/presigned_url_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import (
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestIsSignedRequestWithSignatureParam(t *testing.T) {
+	r := httptest.NewRequest("GET", "https://example.com/example.jpg?OC-Signature=something", nil)
+	if !isSignedRequest(r) {
+		t.Fail()
```
Comment (refs) [inline pkg/middleware/presigned_url_test.go]: wouldn't it be better to have an assertion here? if the test fails we'd rather know why. We can assert on the bool return value 🤷‍♂️

**Review by refs — APPROVED** [review]
👍 LGTM! I think the tests can benefit from a bit of table testing

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ pkg/middleware/presigned_url.go  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-proxy/pullRequest?prid=6040115&bid=19791446)

**Pull request merged**